pkg/lint/rules/deprecations.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright The Helm Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rules // import "helm.sh/helm/v3/pkg/lint/rules"
+
+import "fmt"
+
+// deprecatedApis lists APIs that are deprecated (left) with suggested alternatives (right).
+//
+// An empty rvalue indicates that the API is completely deprecated.
+var deprecatedApis = map[string]string{
+	"extensions/v1 Deployment":             "apps/v1 Deployment",
+	"extensions/v1 DaemonSet":              "apps/v1 DaemonSet",
+	"extensions/v1 ReplicaSet":             "apps/v1 ReplicaSet",
+	"extensions/v1beta1 PodSecurityPolicy": "policy/v1beta1 PodSecurityPolicy",
+	"extensions/v1beta1 NetworkPoliicy":    "networking.k8s.iio/v1beta1 NetworkPolicy",
+	"apps/v1beta1 Deployment":              "apps/v1 Deployment",
+	"apps/v1beta2 Deployment":              "apps/v1 Deployment",
```

Comment:
DaemonSet, ReplicaSet, and StatefulSet all have the situation of being in `extensions/v1beta1`, `apps/v1beta1`, and `apps/v1beta2` prior to now being in `apps/v1`. Shouldn't they all have checks like Deployment does?

Reply:
If you are sure. I read the Kubernetes release notes and they were... unclear. I will add them.

Reply:
This, BTW, is not something I could readily find information about. @bacongobbler basically helped me find release notes that referenced deprecated things, but there appears to be no official documentation for the Kubernetes project. If you happen to find any, please let me know. It's a huge pain to scour release notes for mentions of deprecated APIs.

Reply:
You can see a list in the old reference documentation... https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.15/

This is not documented well. I filed https://github.com/kubernetes/website/issues/20617 to try and deal with the lack of documentation.